io/jvm/src/main/scala/scalaz/effect/RTS.scala

```diff
@@ -273,13 +270,9 @@ private object RTS {
       // was caught but the stack is empty, we make the stack non-empty. This
       // lets us return only the finalizer, which will be null for common cases,
       // and result in zero heap allocations for the happy path.
-      if (caught) {
-        if (stack.isEmpty) stack.push(IdentityCont)
-        (errorHandler, finalizer)
-      } else {
-        finalizer
-      }
+      if (errorHandler ne null) stack.push(errorHandler)
```

Comment:
This is correct: if the error handler has been identified, add it to the stack as the next continuation.

The comment can be changed:

```
// The stack will never be empty if the error was caught, because the error handler will be pushed onto the stack
```